contracts/multisig-prover/src/execute.rs

```diff
@@ -142,42 +142,74 @@ fn get_workers_info(
     })
 }
 
-pub fn update_worker_set(deps: DepsMut, env: Env) -> Result<Response, ContractError> {
-    let config = CONFIG.load(deps.storage)?;
-
-    let workers_info = get_workers_info(&deps, &env, &config)?;
+fn get_worker_sets(deps: &DepsMut, env: &Env, config: &Config) -> (Option<WorkerSet>, WorkerSet) {
```

Comment:
Can we actually just make this `get_next_worker_set`? And it also does the `should_update_worker_set` check, and returns the next worker set, if it needs to be updated?